app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/NetworkTimelineRemoteMediator.kt

```diff
@@ -29,9 +33,13 @@ import retrofit2.HttpException
 @OptIn(ExperimentalPagingApi::class)
 class NetworkTimelineRemoteMediator(
     private val accountManager: AccountManager,
-    private val viewModel: NetworkTimelineViewModel
+    private val viewModel: NetworkTimelineViewModel,
+    db: AppDatabase,
+    private val gson: Gson
```

Comment:
I think all of these can be injected here.

Then you can inject the remote mediator in to the view model. That way the view model doesn't need to receive `db` or `gson` as parameters.

Reply:
The remote mediator has the view model as dependency

(and thus a dependency cylce)